### PR TITLE
feat(hints): add NewHint#34

### PR DESCRIPTION
**Pull request by MegaRedHand**
Closes: #974

## Description

This PR adds the missing hint _NewHint#34_, now known as `SHA256_MAIN_ARBITRARY_INPUT_LENGTH`. Also renames `SHA256_MAIN` to `SHA256_MAIN_CONSTANT_INPUT_LENGTH`. As part of the changes, extract helper function `get_constant_from_var_name` used for retrieving a constant from `constants` `HashMap`.

### CODE REVIEW

File: cairo_programs/sha256_test.cairo [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+    let d = hash[3];
+    assert d = 3665669114;
+    let e = hash[4];
+    assert e = 3297046499;
+    let f = hash[5];
+    assert f = 2052292846;
+    let g = hash[6];
+    assert g = 2424895404;
+    let h = hash[7];
+    assert h = 3807366633;
```
Comment (fmoletta) [inline cairo_programs/sha256_test.cairo:39]:
```suggestion
    assert hash[0] = 3108841401;
    assert hash[1] = 2471312904;
    assert hash[2] = 2771276503;
    assert hash[3] = 3665669114;
    assert hash[4] = 3297046499;
    assert hash[5] = 2052292846;
    assert hash[6] = 2424895404;
    assert hash[7] = 3807366633;
```

File: cairo_programs/sha256_test.cairo [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+    let d = hash[3];
+    assert c = 1331117438;
+    let e = hash[4];
+    assert e = 699003633;
+    let f = hash[5];
+    assert f = 2214481798;
+    let g = hash[6];
+    assert g = 3208491254;
+    let h = hash[7];
+    assert h = 789740750;
```
Comment (fmoletta) [inline cairo_programs/sha256_test.cairo:100]: Similar comment here

File: cairo_programs/sha256_test.cairo [hunk 3]
```diff
@@ -0,0 +1,512 @@
...
+    let d = hash[3];
+    assert d = 0x996fb924;
+    let e = hash[4];
+    assert e = 0x27ae41e4;
+    let f = hash[5];
+    assert f = 0x649b934c;
+    let g = hash[6];
+    assert g = 0xa495991b;
+    let h = hash[7];
+    assert h = 0x7852b855;
```
Comment (fmoletta) [inline cairo_programs/sha256_test.cairo:131]: And here

File: cairo_programs/sha256_test.cairo [hunk 4]
```diff
@@ -0,0 +1,512 @@
...
+    let d = hash[3];
+    assert d = 0x5dae2223;
+    let e = hash[4];
+    assert e = 0xb00361a3;
+    let f = hash[5];
+    assert f = 0x96177a9c;
+    let g = hash[6];
+    assert g = 0xb410ff61;
+    let h = hash[7];
+    assert h = 0xf20015ad;
```
Comment (fmoletta) [inline cairo_programs/sha256_test.cairo:159]: And here

File: cairo_programs/sha256_test.cairo [hunk 5]
```diff
@@ -0,0 +1,512 @@
...
+    let d = hash[3];
+    assert d = 0x0c3e6039;
+    let e = hash[4];
+    assert e = 0xa33ce459;
+    let f = hash[5];
+    assert f = 0x64ff2167;
+    let g = hash[6];
+    assert g = 0xf6ecedd4;
+    let h = hash[7];
+    assert h = 0x19db06c1;
```
Comment (fmoletta) [inline cairo_programs/sha256_test.cairo:203]: ditto

File: cairo_programs/sha256_test.cairo [hunk 6]
```diff
@@ -0,0 +1,512 @@
...
+    let d = hash[3];
+    assert d = 0xb05a6f07;
+    let e = hash[4];
+    assert e = 0x1fc48fa6;
+    let f = hash[5];
+    assert f = 0x6494d085;
+    let g = hash[6];
+    assert g = 0x13b9ba1b;
+    let h = hash[7];
+    assert h = 0x82df20cd;
```
Comment (fmoletta) [inline cairo_programs/sha256_test.cairo:250]: ditto

File: src/hint_processor/builtin_hint_processor/sha256_utils.rs [hunk 7]
```diff
@@ -47,29 +48,42 @@ pub fn sha256_input(
...
-    let mut message: Vec<u8> = Vec::with_capacity(4 * SHA256_INPUT_CHUNK_SIZE_FELTS);
+    // The original code gets it from `ids` in both cases, and this makes it easier
+    // to implement the arbitrary length one
+    let input_chunk_size_felts =
+        get_constant_from_var_name("SHA256_INPUT_CHUNK_SIZE_FELTS", constants)?
+            .to_usize()
+            .unwrap_or(100); // Hack: enough to fail the assertion
+
+    if input_chunk_size_felts >= 100 {
+        return Err(HintError::AssertionFailed("AssertionError".to_string()));
```
Comment (fmoletta) [inline src/hint_processor/builtin_hint_processor/sha256_utils.rs:69]:
```suggestion
        return Err(HintError::AssertionFailed("ids.SHA256_INPUT_CHUNK_SIZE_FELTS == 8".to_string()));
```

File: cairo_programs/sha256_test.cairo (hunk 2 above)
Comment (MegaRedHand) [reply cairo_programs/sha256_test.cairo]: Done!

**Pull request merged**